### DISCUSSION TITLE
Update to Twig 3.15.x

### DISCUSSION THREAD

**Discussion by MoritzLost** [craftcms/cms#16207] [Ideas] [answered]
[Twig 3.15.0](https://github.com/twigphp/Twig/blob/c0689e3e686f5c8b5e9b9bdb10a74ea6ccfea350/CHANGELOG#L5C1-L5C22) brings a bunch of changes and features, most notably the ability to use arrow functions as callables for PHP functions that expect a function argument. This is very useful for collection methods, a lot of which require callables to be used.

Beyond that, there are a couple of deprecations and new syntax options. Upgrading the dependency will allow us to use the new syntax everywhere it's possible, and help us avoid using soon-to-be-deprecated language features.

Craft currently only allows ~3.14.

**Comment by AugustMiller** [comment, marked as answer]
Per Brandon's message in Discord—Twig 3.15 will ship with the upcoming [5.6](https://github.com/craftcms/cms/blob/5.6/CHANGELOG-WIP.md#system) and [4.14](https://github.com/craftcms/cms/blob/4.14/CHANGELOG-WIP.md#system) releases!

**Reply by brandonkelly** [reply]
Craft 4.14.0 and 5.6.0 are out now with Twig 3.15 🎉
